lib/capybara/server.rb

```diff
@@ -4,22 +4,43 @@
 
 module Capybara
   class Server
+    class Counter
+      attr_reader :value
+
+      def initialize(value=0)
+        @value = value
+        @mutex = Mutex.new
+      end
+
+      def increment
+        @mutex.synchronize { @value += 1 }
+      end
+
+      def decrement
+        @mutex.synchronize { @value -= 1 }
+      end
+    end
+
     class Middleware
       attr_accessor :error
 
-      def initialize(app)
+      def initialize(app, counter=Counter.new)
```

Comment:
do we really need to accept a counter here? since it's always passed a new counter in usage couldnt we just create a new counter inside initialize